core/src/main/java/org/jdbi/v3/core/qualifier/QualifiedType.java

```diff
@@ -40,6 +40,7 @@
 public final class QualifiedType<T> {
     private final Type type;
     private final Set<Annotation> qualifiers;
+    private int hashCode;
```

Comment:
May be make it final and calculate in the constructor?

Reply:
I'm going to leave this one as is: making this change seems to regress the benchmarks.  I updated the code to better follow the pattern in String for lazily-computed hashes, assuming the JDK authors know better than I do :)